mllib/src/test/scala/org/apache/spark/ml/feature/RFormulaSuite.scala

```diff
@@ -129,6 +129,23 @@ class RFormulaSuite extends SparkFunSuite with MLlibTestSparkContext with Defaul
     assert(result.collect() === expected.collect())
   }
 
+  test("formula w/o intercept, we should output reference category when encoding string terms") {
```

Comment:
Would also be great to test the impact of the change when there is are interactions present? 
Will this create the same design matrix as R?